go-controller/pkg/clustermanager/userdefinednetwork/template/net-attach-def-template.go

```diff
@@ -185,10 +206,43 @@ func renderCNINetworkConfig(networkName, nadName string, spec SpecGetter) (map[s
 	if netConfSpec.AllowPersistentIPs {
 		cniNetConf["allowPersistentIPs"] = netConfSpec.AllowPersistentIPs
 	}
-
+	if netConfSpec.PhysicalNetworkName != "" {
+		cniNetConf["physicalNetworkName"] = netConfSpec.PhysicalNetworkName
+	}
+	if len(netConfSpec.ExcludeSubnets) > 0 {
+		cniNetConf["excludeSubnets"] = netConfSpec.ExcludeSubnets
+	}
+	if netConfSpec.VLANID != 0 {
+		cniNetConf["vlanID"] = netConfSpec.VLANID
+	}
 	return cniNetConf, nil
 }
 
+func validateLocalnetConfig(cfg userdefinednetworkv1.LocalnetConfig) error {
+	if cfg.Role == userdefinednetworkv1.NetworkRolePrimary {
+		return fmt.Errorf("role can be secondary only")
+	}
+	if cfg.PhysicalNetworkName == "" {
+		return fmt.Errorf("physicalNetworkName cannot be empty")
+	}
+	if len(cfg.PhysicalNetworkName) > 253 {
```

Comment:
same thing for this

Reply:
Please see https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5010/files/082457cd9d6bf3065da210aff8ea2f643b0b4af2..5162882165cb2379c9f36d9a20cac5f679380e47#r2046484060